File: javascript/resistor-color-trio/resistor-color-trio.js
const COLORS = [
  'black',
  'brown',
  'red',
  'orange',
  'yellow',
  'green',
  'blue',
  'violet',
  'grey',
  'white',
];

export class ResistorColorTrio {
  constructor(colors) {
    if (colors.some(c => !COLORS.includes(c))) {
      throw new Error('invalid color');
    }
    const digits = COLORS.indexOf(colors[0]) * 10 + COLORS.indexOf(colors[1]) * 1;
    const zeroes = COLORS.indexOf(colors[2]);
    const value = zeroes === 0 ? digits : digits * (10 ** zeroes);
    this.value = value >= 1000 ? value / 1000 : value;
    this.unit = value >= 1000 ? 'kiloohms' : 'ohms';
  }

  get label() {
    return `Resistor value: ${this.value} ${this.unit}`;
  }
}

File: c/resistor-color-trio/src/resistor_color_trio.c
#include "resistor_color_trio.h"
#include "math.h"

resistor_value_t color_code(resistor_band_t colors[]) {
  int unit = OHMS;
  int value = (colors[0] * 10 + colors[1]) * pow(10, colors[2]);
  if (value % 1000 == 0) {
    value /= 1000;
    unit = KILOOHMS;
  }
  return (resistor_value_t){ .value = value, .unit = (unit_t)unit };
}
